broker/core/multiplexing/src/engine.cc

```diff
@@ -270,12 +278,28 @@ void engine::subscribe(const std::shared_ptr<muxer>& subscriber) {
  *  @param[in] subscriber  Subscriber.
  */
 void engine::unsubscribe_muxer(const muxer* subscriber) {
-  std::lock_guard<std::mutex> l(_engine_m);
+  /* We wait for the events to be sent before removing the muxer. */
+  //  if (!_sending_to_subscribers) {
+  //    SPDLOG_LOGGER_INFO(
+  //        _logger,
+  //        "multiplexing: sending events to muxers before unsubscribe one ; "
+  //        "events to send",
+  //        _kiew.size());
+  //    _sending_to_subscribers = true;
+  std::promise<void> promise;
```

Comment:
is there a risk of asio lock as it can be called by asio layers?

Reply:
I didn't wonder as I took an already existing code...

Reply:
But you're right.

Reply:
After some research, I did not find better than this. I propose to keep it and to improve it when we will have a good idea.